packages/admin/admin/src/mui/menu/CollapsibleItem.tsx

```diff
@@ -1,96 +1,201 @@
+            return "to" in child.props && matchPath(location.pathname, { path: child.props.to, strict: true });
+        }
+        hasSelectedChild.current = false;
+
+        return React.Children.map(children, (child: MenuChild) => {
+            // child is selected
+            if (checkIfPathInLocation(child)) {
+                hasSelectedChild.current = true;
+            }
+
+            // sub child is selected
+            const subChildElements =
+                "children" in child.props ? React.Children.map(child?.props?.children, (child: MenuChild) => child) : ([] as MenuChild[]);
+            if (subChildElements?.some((child: MenuChild) => child.props && checkIfPathInLocation(child))) {
+                hasSelectedChild.current = true;
+            }
+
+            const newItemLevel = itemLevel + 1;
+
+            return React.cloneElement<MenuCollapsibleItemProps | MenuItemRouterLinkProps | MenuItemProps>(child, {
+                level: newItemLevel === 1 || newItemLevel === 2 || newItemLevel === 3 ? newItemLevel : undefined,
+                isMenuOpen,
+                isCollapsibleOpen: isSubmenuOpen,
+            });
         });
-    });
+    }, [children, isMenuOpen, isSubmenuOpen, itemLevel, location.pathname]);
+
+    const closeMenu = () => {
+        setAnchorEl(undefined);
+        setIsSubmenuOpen(false);
+    };
+
+    const handlePopoverOpen = (event: React.MouseEvent<HTMLElement>) => {
+        if (isMenuOpen) return;
+        if (anchorEl !== event.currentTarget) {
+            setAnchorEl(event.currentTarget);
+            setIsSubmenuOpen(true);
+        }
+    };
 
-    const [open, setOpen] = React.useState<boolean>(openByDefault || hasSelectedChild);
+    const handlePopoverClose = (e: React.MouseEvent<HTMLElement>) => {
+        if (isMenuOpen) return;
+        const el = e.target as HTMLElement;
+        const rect = el.getBoundingClientRect();
+        const { clientX, clientY } = e;
 
-    const listClasses = [classes.root];
-    if (hasSelectedChild) listClasses.push(classes.childSelected);
-    if (open) listClasses.push(classes.open);
+        if (childElements?.length && clientX + 2 > rect.right && clientY > rect.top && clientY < rect.bottom) return;
+
+        closeMenu();
+    };
+
+    const listClasses = [];
+    if (hasSelectedChild.current) listClasses.push(classes.childSelected);
+    if (isSubmenuOpen) listClasses.push(classes.open);
+
+    let itemId,
+        mouseOverMenuId = undefined;
+    if (typeof primary === "string") {
+        itemId = `menu-item-${primary?.replace(/\s/g, "-").toLowerCase()}`;
+        mouseOverMenuId = `item-hover-menu-${primary?.replace(/\s/g, "-").toLowerCase()}`;
+    }
 
     return (
-        <div {...otherProps} className={listClasses.join(" ")}>
-            <div className={classes.listItem}>
+        <div {...otherProps}>
```

Comment:
This is missing the `root` class-name. 
Should the `listClasses` not still be applied to this element, instead of the `MenuItem`?

Reply:
✅